src/keypair.rs

```diff
@@ -174,69 +169,131 @@ impl Deref for Key {
 
 // Don't make it implement SizedBytes so that it's not constructible outside of this module.
 impl Key {
-    fn to_arr(&self) -> GenericArray<u8, KeyLen> {
+    fn to_arr<L: ArrayLength<u8>>(&self) -> GenericArray<u8, L> {
```

Comment:
We could potentially store a `GenericArray` in `Key` instead of a `Vec`, now that we have a generic on both `PublicKey` and `PrivateKey`.

Reply:
Done.